app/src/main/java/com/notifier/app/core/data/networking/responseToResult.kt

```
@@ -0,0 +1,34 @@
+ * @param T The expected response type.
+ * @param response The [HttpResponse] received from the network request.
+ * @return A [Result] containing either the parsed response data or a [NetworkError].
+ */
+suspend inline fun <reified T> responseToResult(
+    response: HttpResponse
+): Result<T, NetworkError> {
+    return when (response.status.value) {
+        in 200..299 -> {
+            try {
+                Result.Success(response.body<T>())
+            } catch (e: NoTransformationFoundException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)